File: 0377 - Combination Sum IV/solution.c
int combinationSum4(int* A, int ASz, int target){
    unsigned dp[1001] = {1};
    for(int i = 1; i <= target; i++)
        for(int j = 0; j < ASz; j++)
            if(i >= A[j]) dp[i] += dp[i-A[j]];
    return dp[target];
}

File: 0377 - Combination Sum IV/solution.py
class Solution:
    def combinationSum4(self, A: List[int], target: int) -> int:
        @lru_cache(None)
        def dp(n):
            if n <= 0:
                return int(n == 0)
            return sum(dp(n-i) for i in A)
        return dp(target)